proposed/2021/VisualStudioPackageInstallDeepLink.md

```diff
@@ -0,0 +1,163 @@
+
+## Drawbacks
+
+* Older versions of Visual Studio won't support this.
+* Current proposal is limited to NuGet.org packages - generalizing to packages from any feed will increase engineering complexity significantly as we would need a custom way to isolate the target package in the Browse tab.
+
+## Prior Art
+
+### Download VS Code extension from VS Marketplace
+
+Example Extension: [Dracula Official - Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=dracula-theme.theme-dracula)
+
+Deep link for Dracula Official: [vscode:extension/dracula-theme.theme-dracula](vscode:extension/dracula-theme.theme-dracula)
+
+Behavior: Click the deep link opens VS Code and the extension details page for the target extension, making it a 1 click install experience from there.
+
+![Download Dracula Official from VS Marketplace website](./meta/resources/VisualSutdioPackageInstallDeepLink/Dracula%20Official.png)
+
+## Unresolved Questions
+
+**Q: What should the structure of the deep link URL?**
+
+**Q: If a project level PMUI window is already open or in focus, should we open the package there or open the solution level PMUI anyway for a consistent experience?**
+
+*Pending customer development*
+
+**Q: What happens if NuGet.org is not enabled as a source?**
+
+* Option 1: Open a dialogue explaining to the customer “You are attempting to install a package from NuGet.org but you do not have it enabled as a source. Would you like to enable it or cancel?”
+* Option 2: Switch feed setting to “All” automatically
+
+**Q: What package types should support downloading in Visual Studio (library, template, etc.)?**
+
+Currently, anything but library packages fail to install through the PMUI and don’t make sense to install through the PMUI. 
+
+**Q: What Visual Studio operation can we use as prior art?**
+
+•	Open repo in Visual Studio on GitHub.
+•	Download extension from VS Marketplace website only downloads the VSIX file locally and does not have a VS handler.
+•	Others?
+
+**Q: Can/should we generalize this feature to allow deeplinking of packages from any feed?** This would mean we can’t use the “PackageId:” search filter since it’s NuGet.org specific.
+
+## Resolved Questions
+
+**Q: What happens if the package exists on multiple feeds?**
+
+We follow today’s normal installation behavior, though imperfect, and install the package from whichever feed responds. This is an inherent risk of package version duplication across multiple feeds and addressing the issue is not within the scope of this feature.
+
+We plan to address the dependency confusion issue with the upcoming [package namespaces feature](https://github.com/NuGet/Home/blob/dev/proposed/2021/PackageNamespaces.md).
+
+**Q: What happens if the package is already installed?**
+
+The user flow should proceed exactly as it would if the package were not already installed. This is consistent with today’s behavior where we show installed packages in the Browse tab.
+
```

Comment:
How to pick which NuGet config file should be modified when adding a new source?